src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -346,4 +342,39 @@ public function get_collection_params() {
 		 */
 		return apply_filters( 'rest_pattern_directory_collection_params', $query_params );
 	}
+
+	/*
+	 * Include a hash of the query args, so that different requests are stored in
+	 * separate caches.
+	 *
+	 * MD5 is chosen for its speed, low-collision rate, universal availability, and to stay
+	 * under the character limit for `_site_transient_timeout_{...}` keys.
+	 *
+	 * @link https://stackoverflow.com/questions/3665247/fastest-hash-for-non-cryptographic-uses
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $query_args Query arguments to generate a transient key from.
+	 *
+	 * @return string Transient key.
+	 */
+	protected function get_transient_key( $query_args ) {
+
+		if ( isset( $query_args['slug'] ) ) {
+			// This is an additional precaution because the "sort" function expects an array.
+			$query_args['slug'] = (array) $query_args['slug'];
+
+			// Sort the array so that the transient key doesn't depend on the order of slugs.
+			sort( $query_args['slug'] );
+
+			// Slugs have to be imploded separately as implode doesn't work with recursive arrays.
+			$query_args['slug'] = implode( ',', $query_args['slug'] );
+
+			if ( '' === trim( $query_args['slug'] ) ) {
+				unset( $query_args['slug'] );
+			}
+		}
+
+		return 'wp_remote_block_patterns_' . md5( implode( '-', $query_args ) );
```

Comment:
```suggestion
		return 'wp_remote_block_patterns_' . md5( serialize( $query_args ) );
```

Reply:
Yes, I've been thinking about replacing `implode` with `wp_json_encode` (`json_encode` is faster than `serialize`) in the scope of another PR, but `serialize` will do too.
The result produced by `implode` doesn't depend on the keys of the `$query_args` array. Therefore, this increases the chance of collisions.
Fixed in 2c08eeab216148827b02aa5f99a309948ca31a3b